repos/system_upgrade/el7toel8/actors/cupscheck/libraries/library.py

```diff
@@ -0,0 +1,204 @@
+               'because it is deprecated. The migration script will take '
+               'PrintcapFormat directive from cupsd.conf if used '
+               'and put it into cups-files.conf')
+    if facts.printcap:
+        args = [
+            reporting.Title(title),
+            reporting.Summary(summary),
+            reporting.Tags([reporting.Tags.SERVICES]),
+            reporting.Severity(reporting.Severity.LOW),
+            reporting.RelatedResource('files',
+                                      ('/etc/cups/cupsd.conf '
+                                       '/etc/cups/cups-files.conf'))
+        ]
+
+        report_func(args)
+
+
+def check_env_directives(facts, report_func):
+    """
+    Checks if the data model tells PassEnv/SetEnv directives are used
+    and produces a report.
+
+    :param obj facts: model object containing info about CUPS configuration
+    :param func report_func: creates report
+    """
+    title = ('PassEnv/SetEnv directives are no longer in cupsd.conf')
+    summary = ('The directives were moved into /etc/cups/cups-files.conf '
+               'due security reasons. The migration script will take '
+               'the directives from cupsd.conf if used put them into '
+               '/etc/cups/cups-files.conf')
+    if facts.env:
+        args = [
+            reporting.Title(title),
+            reporting.Summary(summary),
+            reporting.Tags([reporting.Tags.SERVICES]),
+            reporting.Severity(reporting.Severity.MEDIUM),
+            reporting.RelatedResource('files',
+                                      ('/etc/cups/cupsd.conf '
+                                       '/etc/cups/cups-files.conf'))
+        ]
+
+        report_func(args)
+
+
+def check_certkey_directives(facts, report_func):
+    """
+    Checks if the data model tells ServerKey/ServerCertificate directives
+    are used and produces a report.
+
+    :param obj facts: model object containing info about CUPS configuration
+    :param func report_func: creates report
+    """
+    title = ('ServerKey/ServerCertificate directives are substitued '
+             'by ServerKeychain directive')
+    summary = ('The directives were substitued by ServerKeychain directive, '
+               'which now takes a directory as value (/etc/cups/ssl is default). '
+               'The previous directives took a file as value. '
+               'The migration script will copy the files specified in '
+               'directive values into /etc/cups/ssl directory '
+               'if they are not there already')
+    if facts.certkey:
+        args = [
+            reporting.Title(title),
+            reporting.Summary(summary),
+            reporting.Tags([reporting.Tags.SERVICES,
+                            reporting.Tags.AUTHENTICATION]),
+            reporting.Severity(reporting.Severity.HIGH),
+            reporting.RelatedResource('file',
+                                      '/etc/cups/cups-files.conf')
+        ]
+
```

Comment:
note for me: conf group 1. but:
- in case the cert and keys are in different directories, they will be copied into the `/etc/cups/ssl` directory (otherwise no action is needed). In such case, after change of keys or certs, both have to be updated, or what is recommended, to check your setup, modify scripts if needed to use the new directory and remove original ones maually. (msg draft)